langchain4j-mcp-docker/src/test/java/dev/langchain4j/mcp/client/transport/docker/DockerMcpTransportIT.java

```diff
@@ -0,0 +1,117 @@
+package dev.langchain4j.mcp.client.transport.docker;
+
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.mcp.McpToolProvider;
+import dev.langchain4j.mcp.client.DefaultMcpClient;
+import dev.langchain4j.mcp.client.McpClient;
+import dev.langchain4j.mcp.client.transport.McpTransport;
+import dev.langchain4j.model.chat.request.json.JsonStringSchema;
+import dev.langchain4j.service.tool.ToolExecutor;
+import dev.langchain4j.service.tool.ToolProviderResult;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class DockerMcpTransportIT {
+    private static McpClient mcpClient;
+
+    @BeforeAll
+    static void setup() {
+        McpTransport transport = new DockerMcpTransport.Builder()
+                .image("mcp/time")
+                .dockerHost("unix:///var/run/docker.sock")
```

Comment:
I guess we should make this test automatically skipped if this socket does not exist. Otherwise, it will unnecessarily fail in environments where it's not expected to work. For example, on my machine with Podman, the file is `unix:///run/user/1000/podman/podman.sock` (but even after changing the path, the test doesn't work for me). Maybe the path should even be configurable using a system property??

Reply:
I made it conditional on the existance of the Docker socket.